x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/rules_stats_collector.ts

```diff
@@ -65,141 +73,153 @@ const getRulesStatsQuery = (): SearchRequest => ({
   query: {
     match_all: {},
   },
+  runtime_mappings: getIdentifierRuntimeMapping(),
   aggs: {
-    rules: {
+    accounts: {
       terms: {
-        field: 'rule.id',
+        field: 'asset_identifier',
         order: {
           _count: 'desc',
         },
         size: 100,
       },
       aggs: {
-        rule_name: {
-          top_metrics: {
-            metrics: {
-              field: 'rule.name',
-            },
-            size: 1,
-            sort: {
-              '@timestamp': 'desc',
-            },
-          },
-        },
-        rule_section: {
-          top_metrics: {
-            metrics: {
-              field: 'rule.section',
-            },
-            size: 1,
-            sort: {
-              '@timestamp': 'desc',
-            },
-          },
-        },
-        rule_version: {
-          top_metrics: {
-            metrics: {
-              field: 'rule.version',
-            },
-            size: 1,
-            sort: {
-              '@timestamp': 'desc',
-            },
-          },
-        },
-        posture_type: {
-          top_metrics: {
-            metrics: {
-              field: 'rule.benchmark.posture_type',
-            },
-            size: 1,
-            sort: {
-              '@timestamp': 'desc',
+        rules: {
+          terms: {
+            field: 'rule.id',
+            order: {
+              _count: 'desc',
             },
+            size: 100,
           },
-        },
-        rule_number: {
-          top_metrics: {
-            metrics: {
-              field: 'rule.benchmark.rule_number',
+          aggs: {
+            rule_name: {
+              top_metrics: {
+                metrics: {
+                  field: 'rule.name',
+                },
+                size: 1,
+                sort: {
+                  '@timestamp': 'desc',
+                },
```

Comment:
why do we need to sort the results here? isn't the rule name always the same? 

this sorting applied to all the other aggregations here as well, and although I don't think we need to focus on optimizing at this stage, I don't see why we need to sort all those aggregations if we only need 1 of them and we know they all equal the same value, their timestamp shouldn't matter here. 

correct me if I'm wrong, I would love to understand if I'm missing something here

Reply:
you right, I removed the sorting.